lib/cisco_node_utils/bgp_af.rb

```diff
@@ -547,5 +547,183 @@ def redistribute=(should)
+    end
+
+    # route target both auto evpn(Getter/Setter/Default)
+    def route_target_both_auto_evpn
+      config_get('bgp_af', 'route_target_both_auto_evpn', @get_args)
+    end
+
+    def route_target_both_auto_evpn=(enable)
+      @set_args[:state] = (enable ? '' : 'no')
+      config_set('bgp_af', 'route_target_both_auto_evpn', @set_args)
+      set_args_keys_default
+    end
+
+    def default_route_target_both_auto_evpn
+      config_get_default('bgp_af', 'route_target_both_auto_evpn')
+    end
+
+    # route target export
+    def route_target_export
+      cmds = config_get('bgp_af', 'route_target_export', @get_args)
+      # config_get returns an array, so cmds == ['1:1','2:2']
+
+      # normalize to a string
+      cmds.empty? ? default_route_target_export : cmds.join(' ')
+    end
+
+    def route_target_export=(should)
+      # 'should'comes in as a string, normalize into array
+      should = should.nil? ? [] : should.split(' ')
+
+      # 'is' is a string, normalize into array
+      is = route_target_export
+      is = is.nil? ? [] : is.split(' ')
+
+      delta_hash = Utils.delta_add_remove(should, is)
+      puts "delta hash=#{delta_hash}"
+      return if delta_hash.values.flatten.empty?
+      [:add, :remove].each do |action|
+        CiscoLogger.debug('route_target_export delta ' \
+          "#{@get_args}\n #{action}: #{delta_hash[action]}")
+        delta_hash[action].each do |target|
+          state = (action == :add) ? '' : 'no'
+          @set_args[:state] = state
+          @set_args[:community] = target
+
+          config_set('bgp_af', 'route_target_export', @set_args)
+        end
+      end
+    end
+
+    def default_route_target_export
+      config_get_default('bgp_af', 'route_target_export')
+    end
+
+    # route target export_evpn
+    def route_target_export_evpn
+      cmds = config_get('bgp_af', 'route_target_export_evpn', @get_args)
+      # config_get returns an array, so cmds == ['1:1','2:2']
+
+      # normalize to a string
+      cmds.empty? ? default_route_target_export_evpn : cmds.join(' ')
+    end
+
+    def route_target_export_evpn=(should)
+      # 'should'comes in as a string, normalize into array
+      should = should.nil? ? [] : should.split(' ')
+
+      # 'is' is a string, normalize into array
+      is = route_target_export_evpn
+      is = is.nil? ? [] : is.split(' ')
+
+      delta_hash = Utils.delta_add_remove(should, is)
+      puts "delta hash=#{delta_hash}"
+      return if delta_hash.values.flatten.empty?
+      [:add, :remove].each do |action|
+        CiscoLogger.debug('route_target_export_evpn delta ' \
+          "#{@get_args}\n #{action}: #{delta_hash[action]}")
+        delta_hash[action].each do |target|
+          state = (action == :add) ? '' : 'no'
+          @set_args[:state] = state
+          @set_args[:community] = target
+
+          config_set('bgp_af', 'route_target_export_evpn', @set_args)
+        end
+      end
```

Comment:
There's a lot of duplication here with the other route_target setters. I think we could just make a helper function instead, e.g.:

1) replace lines 635 - 648 with:

`route_target_delta(should, is, 'route_target_export_evpn')`

2) Create helper method:

```
def route_target_delta(should, is, prop)
  delta_hash = Utils.delta_add_remove(should, is)
  return if delta_hash.values.flatten.empty?
  [:add, :remove].each do |action|
    CiscoLogger.debug(#{prop} #{@get_args}\n #{action}: #{delta_hash[action]}")
    delta_hash[action].each do |community|
      state = (action == :add) ? '' : 'no'
      @set_args[:state] = state
      @set_args[:community] = community
       config_set('bgp_af', prop, @set_args)
    end
  end
end
```